### PR TITLE
ncurses: don't run strip internally

### PR DESCRIPTION
**Pull request by junghans**
Found here: https://api.travis-ci.org/v3/job/614871616/log.txt

### CODE REVIEW

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def flag_handler(self, name, flags):
 
     def configure(self, spec, prefix):
         opts = [
+            'ac_cv_path_LDCONFIG=true',
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ncurses/package.py]: Can you elaborate what this option does and why do we want it?

Comment (junghans) [reply var/spack/repos/builtin/packages/ncurses/package.py]: The ncurses builds run `ldconfig` after install and then ignores the exit code, spack however still catches that error in the log. By overwriting `ldconfig` to be `true` is doesn't do that.

Using the `LDCONFIG` option would be possible too, but there are some OSes for which it gets overwritten again, hence using the configure internal option `ac_cv_path_LDCONFIG` seems better.

**Comment by junghans** [comment]
> Found here: https://api.travis-ci.org/v3/job/614871616/log.txt

The error is actually:
```
strip:/home/ci/spack/opt/spack/linux-fedora31-haswell/gcc-9.2.1/ncurses-6.1-up3eftkq4gbvsod2pz35hsmtyzyu55hp/bin/tic[.gnu.build.attributes]: corrupt GNU build attribute note: wrong note type: bad value
/usr/bin/install: strip process terminated abnormally
```
Happens on Fedora 31.

**Comment by adamjstewart** [comment]
Need to replace `setup_environment` with `setup_build_environment`

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Ncurses(AutotoolsPackage):
     patch('patch_gcc_5.txt', when='@6.0%gcc@5.0:')
     patch('sed_pgi.patch',   when='@:6.0')
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_build_environment(self, spack_env, run_env):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]: There’s only a single `env` parameter now

Comment (junghans) [reply var/spack/repos/builtin/packages/ncurses/package.py]: Fixed.

**Pull request merged**